### ISSUE TITLE
Pass Vec<Transaction> to indexer handlers?

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/fuel-indexer#104]
- We have 2 different modes of execution on the indexer: wasm, and native
- These 2 modes of execution receive data differently
  - wasm handlers receive an `Entity`
  - native handlers receive a `Receipt` (after https://github.com/FuelLabs/fuel-indexer/pull/90 is merged)
- **Should we unify these APIs by passing in a `Vec<Transaction>` to the handlers?**

Additional context
- If this was done, we'd need to update the examples, as well as the documentation, and book

So for example, this [here](https://github.com/FuelLabs/fuel-indexer/blob/master/indexer/src/service.rs#L166):
```rust
// ....
    next_block = block.height.0 + 1;
        for trans in block.transactions {
             match client.receipts(&trans.id.to_string()).await {
// ...
```
Might become:

```rust
// ....
    let transactions: Vec<Transaction> = block.transactions;
    if let Err(e) = exec.trigger_event(transactions) {
// ....
```

**Comment by ra0x3** [comment]
@tjsharp1
- Wanna get your input on this
- I created https://github.com/FuelLabs/fuel-indexer/issues/103, but this is a smaller more actionable version of what's listed in that issue

**Comment by ra0x3** [comment]
@tjsharp1 
- It might help to include some pseudo code with some of the descriptions/details. (I probably follow that better than plain text)
- I am seeing that in `indexer::service::IndexerService::make_task` the receipts are fetched in the transaction ID fragments via a newly instantiated client [here](https://github.com/FuelLabs/fuel-indexer/blob/master/indexer/src/service.rs#L140)
- Given this, wouldn't that mean that the user would have to use their own `FuelClient` to get the receipts using the `TransactionIdFragment`s?
  - If this is the case, I feel like this would put more work on the user, and passing individual `Receipt`s might be better

**Comment by ra0x3** [comment]
@tjsharp1 
- I know you're working on the SQlite stuff, but I did want to ask how you felt about picking this up when you get some time?
- I spent about a week and a half trying to grok the code generation with macros (not macros in general, but specifically in this case) with no luck 
  - Even had a little experimentation PR https://github.com/FuelLabs/fuel-indexer/pull/91
- My thinking is that rather than me bang my head against the wall with this (and _still_ not get it done), you might be able to pick this up while I work on some of the other issues in parallel 
- Since we'd be doing a few things differently if this issue was completed (handlers would work differently, documentation would have to be updated, etc) this might eventually become a blocker to opening the indexer up
- No pressure obviously, if your plate is full I can reach out and see if we can pull in an extra set of hands 😄

**Comment by ra0x3** [comment]
Awesome, thanks @tjsharp1 🥳 I'll assign this to you and get to work tackling some of the other issues

**Comment by ra0x3** [comment]
@tjsharp1 
- Let me know what you think about this
- After the discussion in the Indexer Sync, I'm thinking maybe it's not a priority to align the WASM/native APIs anymore (at least not right now)?
- I thought it might be a priority back when I was thinking native/wasm would be used equally, but now that I know we're pushing wasm (until the curious dev stumbles upon native once they're comfortable with the service) it seems like it _might_ be a bit of a time sink?
  - You would honestly know this better than I would 
- But the other point you made (i.e., passing transactions to the handlers instead of receipts) I think still stands

**Comment by ra0x3** [comment]
Closing as completed in #178
